src/components/Avatar/Avatar.tsx

```diff
@@ -0,0 +1,123 @@
+import clsx from 'clsx';
+import React from 'react';
+import Icon from '../Icon';
+import styles from './Avatar.module.css';
+
+export type UserData = {
+  /**
+   * The full name of the attached user (e.g., Jane Doe, David S. Pumpkins)
+   */
+  fullName: string;
+  /**
+   * User ID associated with the attached user
+   */
+  id?: string | number;
+  /**
+   * Additional data for an attached user (email, etc.)
+   */
+  [k: string]: string | number | undefined;
+};
+
+export type AvatarElement = 'div' | 'button';
+
+export interface Props {
+  /**
+   * Label for the given avatar. Defaults to a string using user data.
+   */
+  ariaLabel: string;
+  as?: AvatarElement;
+  /**
+   * CSS class names that can be appended to the component.
+   */
+  className?: string;
+  /**
+   * The shape of the avatar
+   */
+  shape?: 'circle' | 'square';
+  /**
+   * The size of the component
+   */
+  size?: 'sm' | 'md' | 'lg';
+  /**
+   * The URL to an image resource (loaded lazily)
+   */
+  src?: string;
+  /**
+   * The user associated with this avatar
+   */
+  user?: UserData;
+  /**
+   * Variants of how the avatar will be portrayed
+   */
+  variant?: 'icon' | 'initials' | 'image';
+}
+
+function getInitials(fromName: string): string {
+  /**
+   * Scenarios:
+   * - User's name is spelled as first and last name: John Smith
+   * - User's name has a middle name or initial: John C. Smith
+   * - User's Name has dashes in it
+   */
+  return fromName
+    .split(' ')
+    .map((part) => part[0])
+    .reduce(
+      (prev, curr, idx, arr) =>
+        idx === 0 || idx === arr.length - 1 ? prev + curr : prev,
+      '',
+    )
+    .toUpperCase();
+}
+
+/**
+ * BETA: This component is still a work in progress and is subject to change.
+ *
+ * `import {Avatar} from "@chanzuckerberg/eds";`
+ *
+ * Representation of a single, unique user, keyed by the user name
+ */
+export const Avatar = ({
+  ariaLabel,
+  as: TagName = 'div' as const,
+  className,
+  shape = 'circle',
+  size = 'md',
+  user,
+  variant = 'initials',
+  src,
+  ...other
+}: Props) => {
+  const componentClassName = clsx(
+    styles['avatar'],
+    shape === 'circle' && styles['avatar--circle'],
+    shape === 'square' && styles['avatar--square'],
+    size === 'sm' && styles['avatar--sm'],
+    size === 'md' && styles['avatar--md'],
+    size === 'lg' && styles['avatar--lg'],
+    variant === 'icon' && styles['avatar--icon'],
+    variant === 'initials' && styles['avatar--initials'],
+    variant === 'image' && styles['avatar--image'],
```

Comment:
Alternatively ```size && styles[`avatar--${size}`]``` and for variant. Not sure what is more readable tho lol

Reply:
oh i like this. unwritten lines of code never have bugs 🐛